osquery/tables/applications/posix/prometheus_metrics.cpp

```diff
@@ -0,0 +1,112 @@
+/*
+ *  Copyright (c) 2014-present, Facebook, Inc.
+ *  All rights reserved.
+ *
+ *  This source code is licensed under the BSD-style license found in the
+ *  LICENSE file in the root directory of this source tree. An additional grant
+ *  of patent rights can be found in the PATENTS file in the same directory.
+ *
+ */
+#include <sstream>
+
+#include <osquery/config.h>
+#include <osquery/logger.h>
+#include <osquery/tables.h>
+
+#include <osquery/tables/applications/posix/prometheus_metrics.h>
+#include <osquery/tables/applications/posix/prometheus_metrics_utils.h>
+
+namespace http = boost::network::http;
+
+namespace osquery {
+namespace tables {
+
+void PrometheusMetrics::parseScrapeResults() {
+  for (auto const& target : scrapeResults_) {
+    std::stringstream ss(target.second->content);
+    std::string dest;
+    std::string ts(std::to_string(target.second->timestampMS.count()));
+
+    while (std::getline(ss, dest)) {
+      if (dest[0] != '#') {
+        std::stringstream iss(dest);
+        std::string idest;
+        std::vector<std::string> metric;
+
+        while (std::getline(iss, idest, ' ')) {
```

Comment:
[We have a helper function](https://github.com/facebook/osquery/blob/master/osquery/core/conversions.h#L64) `osquery::split` that might better suite what you're trying to do here and would save on local variables. Then you could change your loop to look something like:
`for(const auto& tok : osquery::split(dest, ' '))`